vck-aries/src/commonTest/kotlin/at/asitplus/wallet/lib/aries/IssueCredentialMessengerTest.kt

```diff
@@ -4,22 +4,20 @@ import at.asitplus.wallet.lib.agent.Holder
 import at.asitplus.wallet.lib.agent.HolderAgent
 import at.asitplus.wallet.lib.agent.Issuer
 import at.asitplus.wallet.lib.agent.IssuerAgent
-import at.asitplus.wallet.lib.agent.KeyPairAdapter
-import at.asitplus.wallet.lib.agent.RandomKeyPairAdapter
+import at.asitplus.wallet.lib.agent.KeyWithCert
+import at.asitplus.wallet.lib.agent.EphemeralKeyWithSelfSignedCert
 import at.asitplus.wallet.lib.agent.SubjectCredentialStore
 import at.asitplus.wallet.lib.data.AtomicAttribute2023
 import at.asitplus.wallet.lib.data.ConstantIndex
 import com.benasher44.uuid.uuid4
 import io.kotest.core.spec.style.FreeSpec
-import io.kotest.matchers.collections.shouldBeEmpty
-import io.kotest.matchers.collections.shouldNotBeEmpty
 import io.kotest.matchers.shouldBe
 import io.kotest.matchers.types.shouldBeInstanceOf
 
 class IssueCredentialMessengerTest : FreeSpec() {
 
-    private lateinit var issuerKeyPair: KeyPairAdapter
-    private lateinit var holderKeyPair: KeyPairAdapter
+    private lateinit var issuerKeyPair: KeyWithCert
```

Comment:
Does this cover the option, that `certificate` is optional?

Reply:
KeyWithCert has the cert optional, but generates a self-signed one by default. if you have a better name, let me know

Reply:
see below